FlowCrypt/src/main/java/com/flowcrypt/email/ui/activity/fragment/CreateMessageFragment.kt

```diff
@@ -1233,6 +1197,12 @@ class CreateMessageFragment : BaseFragment<FragmentCreateMessageBinding>(),
   private fun setupComposeMsgViewModel() {
     lifecycleScope.launchWhenStarted {
       composeMsgViewModel.messageEncryptionTypeStateFlow.collect {
+        composeMsgViewModel.updateOutgoingMessageInfo(
+          composeMsgViewModel.outgoingMessageInfoStateFlow.value.copy(
```

Comment:
These 4 lines seem to repeat themselves in this file. Could they be done as a method?

Reply:
I'm not sure what will be better - repeat these lines or create different methods in ComposeMsgViewModel.

I have the following code

```kotlin
fun updateOutgoingMessageInfo(outgoingMessageInfo: OutgoingMessageInfo) {
    outgoingMessageInfoMutableStateFlow.update { outgoingMessageInfo }
}
```

where

```kotlin
data class OutgoingMessageInfo constructor(
  val account: String? = null,
  val subject: String? = null,
  val msg: String? = null,
  val toRecipients: List<InternetAddress>? = null,
  val ccRecipients: List<InternetAddress>? = null,
  val bccRecipients: List<InternetAddress>? = null,
  val from: InternetAddress? = null,
  val atts: List<AttachmentInfo>? = null,
  val forwardedAtts: List<AttachmentInfo>? = null,
  val encryptionType: MessageEncryptionType? = null,
  @MessageType val messageType: Int = MessageType.NEW,
  val replyToMsgEntity: MessageEntity? = null,
  val uid: Long = 0,
  val password: CharArray? = null
)
```

Every field in OutgoingMessageInfo can be updated.

For example, to update `subject` I can create a new method like this one

```kotlin
fun updateSubject(newSubject: String? = null) {
    outgoingMessageInfoMutableStateFlow.update {
      it.copy(subject = newSubject)
    }
 }
```
But for other fields, I have to create similar methods. What about a case when I need to update a few parameters?

For example, this code

```kotlin
        val forwardedAtts = attachments.filter { it.id != null && it.isForwarded }
        composeMsgViewModel.updateOutgoingMessageInfo(
          composeMsgViewModel.outgoingMessageInfoStateFlow.value.copy(
            atts = attachments - forwardedAtts.toSet(),
            forwardedAtts = forwardedAtts
          )
        )
```
should have something like this one

```kotlin
fun updateAttsAndForwardedAtts(
    atts: List<AttachmentInfo>? = null,
    forwardedAtts: List<AttachmentInfo>? = null
  ) {
    outgoingMessageInfoMutableStateFlow.update {
      it.copy(atts = atts, forwardedAtts = forwardedAtts)
    }
  }
```

More combinations -> more methods

This code looks like repetitions but it's flexible. (To be honest, it's not a repetition, it's Kotlin usage. I can't or don't know how to simplify it. It's already a common case).
```kotlin
composeMsgViewModel.updateOutgoingMessageInfo(
          composeMsgViewModel.outgoingMessageInfoStateFlow.value.copy(
            param = newParap,
            .....
          )
        )
```

Reply:
understood